src/routes/docs/tutorials/nextjs/step-4/+page.markdoc

```diff
@@ -0,0 +1,219 @@
+      <div className="u-max-width-350	u-width-full-line u-padding-8">
+        <h1 className="heading-level-1 font-bold">Sign Up</h1>
+        <p className="body-text-1 u-bold u-padding-block-16">
+          Already have an account?{" "}
+          <Link href="/login" className="u-cursor-pointer u-underline">
+            Login
+          </Link>
+        </p>
+        <form onSubmit={signup}>
+          <div>
+            <label className="label">Name</label>
+            <div className="input-text-wrapper">
+              <input
+                className="input-text"
+                placeholder="Name"
+                id="name"
+                type="text"
+                required={true}
+                onChange={(e) => setName(e.target.value)}
+              />
+            </div>
+          </div>
+          <div>
+            <label className="label">Email</label>
+            <div className="input-text-wrapper">
+              <input
+                className="input-text"
+                placeholder="Email"
+                id="email"
+                type="email"
+                required={true}
+                onChange={(e) => setEmail(e.target.value)}
+              />
+            </div>
+          </div>
+          <div>
+            <label className="label">Password</label>
+            <div className="input-text-wrapper">
+              <input
+                className="input-text"
+                placeholder="Password"
+                id="password"
+                type="password"
+                required={true}
+                onChange={(e) => setPassword(e.target.value)}
+              />
+            </div>
+          </div>
+
+          <div className="u-padding-block-16">
+            <button
+              type="submit"
+              disabled={!email || !password}
+              className="button"
+            >
+              Sign Up
+            </button>
+          </div>
+        </form>
+      </div>
+    </section>
+  );
+};
+
+export default SignUp;
+```
+
+Now let's add the signup function. We will be using Appwrite's `create` method to create a new user. We will also be using `react-hot-toast` to show toast messages.
+
+
+```js
+const signup = async (e) => {
+  e.preventDefault();
+  try {
+    await account.create(ID.unique(), email, password, name);
+    await account.createEmailSession(email, password);
+    const jwt = await account.createJWT();
+    const user = await account.get();
+    cookie.set("jwt", jwt.jwt);
+    cookie.set("userid", user.$id);
+    toast.success("Signed up successfully");
+    router.push("/");
+  } catch (error) {
+    toast.error(error.message);
+  }
+};
+```
+
+After the user is created, we are creating a session using `createEmailSession` method. Then we are creating a JWT token using `createJWT` method. We are storing the JWT token and user id in the cookie. Then we are redirecting the user to the home page.
```

Comment:
I think this can be better explained, e.g. why are we doing these things?

```suggestion
First, we need to create the user in our Appwrite instance. After the user is created, we create a session using `createEmailSession` method, to effectively login the user into our app. We then create a JWT token using the `createJWT` method, and store the token and user ID into a cookie. This allows the user to remain logged in, even if the app is closed.

With all that done, we can redirect the user to the home page.
```